dsymbol/src/dsymbol/ufcs.d

```diff
@@ -257,58 +257,72 @@ private DSymbol*[] getUFCSSymbolsForParenCompletion(const(DSymbol)* symbolType,
 
 }
 
-private bool willImplicitBeUpcasted(const(DSymbol)* from, const(DSymbol)* to)
+private bool willImplicitBeUpcasted(ref const(DSymbol) incomingSymbolType, ref const(DSymbol) significantSymbolType)
```

Comment:
I think we should use `scope ref` everywhere to indicate that we don't store the function or any of its children outside the function.